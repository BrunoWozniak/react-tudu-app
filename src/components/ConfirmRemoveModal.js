import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';

export default class ConfirmRemoveModal extends React.Component {
    constructor(props) {
        super(props);
    }
    
    render () {
        return(
            <Modal
                isOpen={!!this.props.confirmDelete}
                contentLabel="Are you sure?"
                closeTimeoutMS={200}
                className="modal"
            >
                <h3 className="modal__title">Are you sure?</h3>
                <p className="modal__body">Please confirm you want to delete this expense.</p>
                <div className="modal__buttons">
                    <button className="button" id="btnRemoveConfirm" onClick={this.props.onConfirmDelete}> Confirm </button>
                    <button className="button button--secondary" id="btnRemoveCancel" onClick={this.props.onCancelDelete} > Cancel </button>
                </div>
            </Modal>
        );
    }
}