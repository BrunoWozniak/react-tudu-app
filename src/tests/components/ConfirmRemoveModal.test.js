import React from 'react';
import { shallow } from 'enzyme';
import Modal from 'react-modal';
import ConfirmRemoveModal from '../../components/ConfirmRemoveModal';

let onCancelDelete, onConfirmDelete, wrapper;

beforeAll(() => {
    onCancelDelete = jest.fn();
    onConfirmDelete = jest.fn();
});

test('should render ConfirmRemoveModal correctly', () => {
    wrapper = shallow(<ConfirmRemoveModal
        confirmDelete={true}
        onCancelDelete={onCancelDelete}
        onConfirmDelete={onConfirmDelete}
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ConfirmRemoveModal correctly (not open)', () => {
    wrapper = shallow(<ConfirmRemoveModal
        confirmDelete={false}
        onCancelDelete={onCancelDelete}
        onConfirmDelete={onConfirmDelete}
    />);
    expect(wrapper).toMatchSnapshot();
});

test('should handle onCancel', () => {
    wrapper = shallow(<ConfirmRemoveModal
        confirmDelete={true}
        onCancelDelete={onCancelDelete}
        onConfirmDelete={onConfirmDelete}
    />);
    wrapper.find({ id: 'btnRemoveCancel' }).simulate('click');
    expect(onCancelDelete).toHaveBeenCalled();
});

test('should handle onConfirm', () => {
    wrapper = shallow(<ConfirmRemoveModal
        confirmDelete={true}
        onCancelDelete={onCancelDelete}
        onConfirmDelete={onConfirmDelete}
    />);
    wrapper.find({ id: 'btnRemoveConfirm' }).simulate('click');
    expect(onConfirmDelete).toHaveBeenCalled();
});