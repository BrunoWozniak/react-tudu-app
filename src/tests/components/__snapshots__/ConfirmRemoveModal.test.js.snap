// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmRemoveModal correctly (not open) 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Are you sure?"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
>
  <h3
    className="modal__title"
  >
    Are you sure?
  </h3>
  <p
    className="modal__body"
  >
    Please confirm you want to delete this expense.
  </p>
  <div
    className="modal__buttons"
  >
    <button
      className="button"
      id="btnRemoveConfirm"
      onClick={[Function]}
    >
       Confirm 
    </button>
    <button
      className="button button--secondary"
      id="btnRemoveCancel"
      onClick={[Function]}
    >
       Cancel 
    </button>
  </div>
</Modal>
`;

exports[`should render ConfirmRemoveModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Are you sure?"
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
>
  <h3
    className="modal__title"
  >
    Are you sure?
  </h3>
  <p
    className="modal__body"
  >
    Please confirm you want to delete this expense.
  </p>
  <div
    className="modal__buttons"
  >
    <button
      className="button"
      id="btnRemoveConfirm"
      onClick={[Function]}
    >
       Confirm 
    </button>
    <button
      className="button button--secondary"
      id="btnRemoveCancel"
      onClick={[Function]}
    >
       Cancel 
    </button>
  </div>
</Modal>
`;
